pkg/controller/kieapp/defaults/defaults.go

```diff
@@ -131,33 +131,51 @@ func GetEnvironment(cr *api.KieApp, service kubernetes.PlatformService) (api.Env
 }
 
 func setProductLabels(cr *api.KieApp, env *api.Environment) {
-	setObjectLabels(cr, &env.Console, "business-central")
+	setObjectLabels(cr, &env.Console, "business-central",  getSubComponentTypeByImageName(constants.BcImageName))
+	envConstants, _ := constants.EnvironmentConstants[cr.Status.Applied.Environment] // this will be replaced with the set ObejctLAbel with kie-server component
 	for index := range env.Servers {
-		setObjectLabels(cr, &env.Servers[index], "kie-server")
+		//this will be replaced, the kie-server will be replaced by other components
+		setObjectLabels(cr, &env.Servers[index], "kie-server",  getSubComponentTypeByImageName(envConstants.App.ImageName))
 	}
-	setObjectLabels(cr, &env.SmartRouter, "smart-router")
-	setObjectLabels(cr, &env.ProcessMigration, "process-migration")
+	setObjectLabels(cr, &env.SmartRouter, "smart-router",  getSubComponentTypeByImageName(constants.SmartRouterImageName))
+
+	//this is supposed to stay here as previous but with type application
+	setObjectLabels(cr, &env.ProcessMigration, "process-migration", getSubComponentTypeByImageName(constants.ProcessMigrationImageName))
+}
+
+func getSubComponentTypeByImageName(imageName string) string {
+	retValue := "application"
+	if imageName == constants.SmartRouterImageName || imageName == constants.ControllerImageName {
+		retValue = "infrastructure"
+	}
+	return retValue
 }
 
-func setObjectLabels(cr *api.KieApp, object *api.CustomObject, component string) {
+
+func setObjectLabels(cr *api.KieApp, object *api.CustomObject, component string, subcomponentType string) {
```

Comment:
no longer need to accept `subcomponentType` here

Reply:
correct, fixed